### ISSUE TITLE
Exception thrown when using BatchTypeExtension and Execution Metrics

### ISSUE THREAD

**Issue by rafael-zilberman** [graphql-aspnet/graphql-aspnet#29]
I have followd the documentation upon using `BatchTypeExtension` to get nested many-to-many results and got the following exception:
`Unhandled Exception | Type: 'GraphTypeDeclarationException', Message: 'Graph Type Mismatch, List<Bakery>. Collections and KeyValuePair enumerable types cannot be directly searched. Instead, search for the types supplied to the generic type declaration.`

The code I am using is the code from the example in the documentation: 
```
public class BakedGoodsCompanyController : GraphController
{
    [QueryRoot("bakeries")]
    public async Task<List<Bakery>> RetrieveBakeries(Region region){/*...*/}

    // declare the batch operation as an extension
    [BatchTypeExtension(typeof(Bakery), "orders", typeof(List<CakeOrder>))]
    public async Task<IGraphActionResult> RetrieveCakeOrders(
        IEnumerable<Bakery> bakeries,
        int limitTo = 15)
    {
        //fetch all the orders at once
        var allOrders = await _service.RetrieveCakeOrders(bakeries.Select(x => x.Id), limitTo);

        // return the batch of orders
        return this.StartBatch()
            .FromSource(bakeries, bakery => bakery.Id)
            .WithResults(allOrders, cakeOrder => cakeOrder.BakeryId)
            .Complete();
    }
}
```

I couldn't find any place that uses the type mentioned in the exception above.
Also the nuget package did not contain the debugging symbols and I couldn't see what the middleware that throwing this exception actually does.
After I cloned the projected and built it wil in debug mode to generate the debugging symbols I was able to determine that the exception is thrown while the `ApolloTracingMatricsV1` is trying to Generate Execution Result and to write down the request DataSource at line 191.

**Comment by kevin-carroll** [comment]
Thanks for the detailed write up @rafael-zilberman !

I've been able to reproduce this with the exact error message you indicated. It looks like the tracing metrics are trying to infer graph types from the resultant data objects for reporting.  By their nature, the data from a batch operation doesn't fit into the schema (since its an intermediate object) so the process breaks down and the erroneous exception you encountered is thrown. 

This shouldn't be too difficult of a fix. I'll get it triaged this week and if nothing gets in the way I should have a fix out this weekend.

Cheers,
Kevin

**Comment by rafael-zilberman** [comment]
That's would be great!

Also to ease up the debugging process (especially of a package in alpha version) it will be much appreciated if the publish process will also upload the debugging symbols to nuget symbols server.

Thank you for your time,
Rafael.

**Comment by kevin-carroll** [comment]
Hey @rafael-zilberman ,

In regards to your comment about the symbols, they are included on nuget. You can see them on the nuget package page on the right: https://www.nuget.org/packages/GraphQL.AspNet/   under `Download Symbols`.

It does require you to enable enable Nuget's Symbol server in Visual Studio:
`Tools -> Options -> Debugging -> Symbols -> [Check Nuget.org Symbol Server]`
AND
`Tools -> Options -> Debugging -> General -> [Uncheck 'Enable Just My Code']`

I just spun up a fresh new VM and was able to pull download and step through the code as expected with a test project after setting these two options (they are turned off by default).  

Are you looking for another workflow for getting symbols locally? Can you describe your expectations and how you may have seen it accomplished in other packages; I'll see what I can do to accommodate the request.

Cheers,
KC

**Comment by kevin-carroll** [comment]
Fixed with release [v0.6.3](https://github.com/graphql-aspnet/graphql-aspnet/releases/tag/v0.6.3-beta)
